Lib/tarfile.py

```diff
@@ -2509,28 +2539,53 @@ def main():
     elif args.create is not None:
         tar_name = args.create.pop(0)
         _, ext = os.path.splitext(tar_name)
-        compressions = {
-            # gz
-            '.gz': 'gz',
-            '.tgz': 'gz',
-            # xz
-            '.xz': 'xz',
-            '.txz': 'xz',
-            # bz2
-            '.bz2': 'bz2',
-            '.tbz': 'bz2',
-            '.tbz2': 'bz2',
-            '.tb2': 'bz2',
-        }
         tar_mode = 'w:' + compressions[ext] if ext in compressions else 'w'
         tar_files = args.create
+        if args.verbose:
+            print('Create {!r}:'.format(tar_name))
+        # Actual adding files is done later.
+    elif args.add is not None:
+        tar_name = args.add.pop(0)
+        _, ext = os.path.splitext(tar_name)
 
-        with TarFile.open(tar_name, tar_mode) as tf:
-            for file_name in tar_files:
-                tf.add(file_name)
-
+        tar_mode = 'a'
+        tar_files = args.add
         if args.verbose:
-            print('{!r} file created.'.format(tar_name))
+            print('Add to {!r}:'.format(tar_name))
+
+        if ext in compressions:
+            parser.exit(
+                1,
+                "Error: Add is not a supported operation for compressed tar files")
+
+        if os.path.exists(tar_name) and not is_tarfile(tar_name):
+            parser.exit(
+                1,
+                "Error: {!r} exists but is not a tar file so cannot add".format(
+                    tar_name))
+
+    if tar_name and tar_mode:  # Add or Create so add files
+        import glob
+        if len(tar_files):
+            with TarFile.open(tar_name, tar_mode) as tf:
+                for file_name in tar_files:
+                    if args.verbose:
+                        print('  Add: {!r}'.format(file_name))
+                    if os.path.exists(file_name):
+                        tf.add(file_name)
+                    else:
+                        # Not a file or dir so possible wildcard
+                        if args.verbose:
+                            print("Possible  Wildcard Add!", file_name)
```

Comment:
I personally don't mind and the other verbose outputs were to stdout so I left them the same. Normally I only separate them when there is a chance that the stdout messages are going to be piped to another process but not when they are unlikely to be piped.